common/src/messaging/crossbeam_messaging.rs

```diff
@@ -0,0 +1,75 @@
+use std::fmt::Debug;
+
+use async_trait::async_trait;
+
+use super::{AsyncReceiver, AsyncSender};
+
+#[derive(Debug)]
+pub enum CrossbeamSenderError {
+    ChannelDisconnected,
+}
+
+impl std::error::Error for CrossbeamSenderError {}
+
+impl std::fmt::Display for CrossbeamSenderError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            CrossbeamSenderError::ChannelDisconnected => write!(f, "Channel was disconnected"),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum CrossbeamReceiverError {
+    ReceiveError(crossbeam_channel::RecvError),
+}
+
+impl std::error::Error for CrossbeamReceiverError {}
+
+impl std::fmt::Display for CrossbeamReceiverError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            CrossbeamReceiverError::ReceiveError(e) => write!(f, "ReceiveError {}", e),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct CrossbeamReceiver<T: Send> {
+    receiver: crossbeam_channel::Receiver<T>,
+}
+
+#[derive(Clone)]
+pub struct CrossbeamSender<T: Send> {
+    sender: crossbeam_channel::Sender<T>,
+}
+
+impl<T: Send> CrossbeamReceiver<T> {
+    pub fn new(receiver: crossbeam_channel::Receiver<T>) -> Self {
+        Self { receiver }
+    }
+}
+
+#[async_trait]
```

Comment:
we could try with nightly also ^^

Reply:
Why not, but I'm not sure about introducing nightly features, knowing that some contributors are not that fluent with rust and its toolchain. This might need further discussing with the others ?